sdk/storage_blobs/src/options/tags.rs

```diff
@@ -28,67 +45,68 @@ impl Tags {
     pub fn new() -> Self {
         Self::default()
     }
-
-    pub fn insert<S>(&mut self, key: S, value: S)
+    pub fn insert<K, V>(&mut self, key: K, value: V)
     where
-        S: Into<String>,
+        K: Into<String>,
+        V: Into<String>,
     {
-        if self.tag_set.is_none() {
-            self.tag_set = Some(TagSet { tag: vec![] });
-        }
-
-        let tag_set = self.tag_set.as_mut().unwrap();
-        tag_set.tag.push(Tag {
+        self.tag_set.tag.push(Tag {
             key: key.into(),
             value: value.into(),
         });
     }
 
-    pub fn to_xml(&self) -> String {
-        let mut s = String::new();
-        s.push_str("<?xml version=\"1.0\" encoding=\"utf-8\"?>\n<Tags>\n<TagSet>\n");
-        if let Some(tag_set) = &self.tag_set {
-            for tag in &tag_set.tag {
-                let node = format!(
-                    "\t<Tag><Key>{}</Key><Value>{}</Value></Tag>\n",
-                    tag.key, tag.value
-                );
-                s.push_str(&node);
-            }
+    pub fn to_xml(&self) -> azure_core::Result<String> {
+        let mut s = String::from("<?xml version=\"1.0\" encoding=\"utf-8\"?><Tags><TagSet>");
+        for tag in &self.tag_set.tag {
+            write!(
+                &mut s,
+                "<Tag><Key>{}</Key><Value>{}</Value></Tag>",
+                tag.key, tag.value
+            )
+            .context(ErrorKind::DataConversion, "failed to write Tags xml")?;
```

Comment:
Just as an FYI: the call can't fail here (writing to a `String` always works), so `unwrap` would be arguably more appropriate here.